asl_xdsl/dialects/asl.py

```diff
@@ -407,125 +349,41 @@ def __init__(
         )
 
 
-@irdl_op_definition
-class NotOp(IRDLOperation):
-    """A bitwise NOT operation."""
-
-    name = "asl.not_bool"
-
-    arg = operand_def(BoolType())
-    res = result_def(BoolType())
-
-    assembly_format = "$arg attr-dict"
-
-    def __init__(self, arg: SSAValue, attr_dict: Mapping[str, Attribute] = {}):
-        super().__init__(
-            operands=[arg],
-            result_types=[BoolType()],
-            attributes=attr_dict,
-        )
-
-
-@irdl_op_definition
-class BoolToI1Op(IRDLOperation):
-    """A hack to convert !asl.bool to i1 so that we can use scf.if."""
-
-    name = "asl.bool_to_i1"
+class UnaryIntOp(IRDLOperation):
```

Comment:
nit:
```suggestion
class UnaryIntOp(IRDLOperation, abc.ABC):
```
I've found it quite useful to leverage Python's `abc` module to let both the type-checker and the runtime enforce that classes designed to be abstract aren't instantiated directly, and that abstract methods are implemented by subclasses.

Reply:
Sounds good - adding that in an extra commit to this PR